src/contexts/navigation/GalleryNavigationProvider.tsx

```diff
@@ -21,6 +22,12 @@ type Props = {
 
 export function GalleryNavigationProvider({ children }: Props) {
   const [historyStackLength, setHistoryStackLength] = useState(0);
+  const { asPath } = useRouter();
+  const [historyStack, setHistoryStack] = useState<string[]>([]);
+
+  useEffect(() => {
+    setHistoryStack((stack) => [...stack, asPath]);
+  }, [asPath]);
```

Comment:
will this keep pushing to the stack even when clicking "back"? should we only add to the stack in the `pushState` function below?

Reply:
i think we should remove this effect and replace with two things:
1. add `setHistoryStack((stack) => [...stack, asPath]);` in the `pushState` function below
2. add `setHistoryStack((stack) => stack.slice(0, -1));` in the `popState` function below

Reply:
Definitely, I started down this road at first with the intention of ideally tracking the browser history 1:1 but it kept getting out of sync if I navigated around and clicked the back and forward buttons a bunch. When it got out of sync, the "Back to Gallery" button would navigate me to all sorts of weird places.

Then I thought that since all I wanted to do was know what url I was on previously, I could instead just have a linear list of urls I've been to. 
So lets say I navigate A -> B then hit the back button.
Browser history would just show [A] because it added B when I went to B, but then popped it when I clicked back.
My list would show [A,B,A] and that's just a direct, dumb representation of where you've been. 

My reasoning was that
- This list isn't going to be used to navigate forwards or backwards in history, but instead just to see what pages the user came from.
- Since we're not trying to recreate the browser history 1:1, this has less risk of getting out of sync and causing weird behavior

I can rename this and add a comment as well to clarify if the above makes sense to you